crt0stack/src/lib.rs

```diff
@@ -0,0 +1,529 @@
+                    value: v as _,
+                });
+            }
+            AuxvEntry::Notelf(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_NOTELF,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::Uid(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_UID,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::EUid(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_EUID,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::Gid(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_GID,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::EGid(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_EGID,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::Platform(v) => {
+                this.platform.replace(v);
+            }
+            AuxvEntry::HWCap(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_HWCAP,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::ClockTick(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_CLKTCK,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::Secure(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_SECURE,
+                    value: if v { 1 } else { 0 },
+                });
+            }
+            AuxvEntry::BasePlatform(v) => {
+                this.base_platform.replace(v);
+            }
+            AuxvEntry::Random(v) => {
+                this.random.replace(v);
+            }
+            AuxvEntry::HWCap2(v) => {
+                this = this.add_auxv_pair(AuxvPair {
+                    key: AT_HWCAP2,
+                    value: v as _,
+                });
+            }
+            AuxvEntry::ExecFilename(v) => {
+                this.exec_fn.replace(v);
+            }
+        }
+        this
+    }
+
+    /// serialize to a mutable byte slice
+    ///
+    /// For more details see a [LWN article](https://lwn.net/Articles/631631/)
+    #[inline]
+    pub fn serialize(self, slice: &mut [u8]) -> Result<usize, Error> {
+        let mut sp = Stack::new(slice);
+        let mut str_ptr: [*const u8; SIZE] = [core::ptr::null(); SIZE];
+        let mut str_len: usize = 0;
+        let mut rand_ptr: Option<*const u8> = None;
+        let mut platform_ptr: Option<*const u8> = None;
+        let mut base_platform_ptr: Option<*const u8> = None;
+        let mut exec_fn_ptr: Option<*const u8> = None;
+
+        str_ptr[str_len] = core::ptr::null();
+        str_len += 1;
```

Comment:
Would this change be equivalent and let us accomplish it when the variable is initialized since I think the way `str_ptr` is declared ensures it's populated with `core::ptr::null()` regardless.

```diff
diff --git a/crt0stack/src/lib.rs b/crt0stack/src/lib.rs
index 15398ed..564bb84 100644
--- a/crt0stack/src/lib.rs
+++ b/crt0stack/src/lib.rs
@@ -240,15 +240,12 @@ impl<'a> Crt0Stack<'a> {
     pub fn serialize(self, slice: &mut [u8]) -> Result<usize, Error> {
         let mut sp = Stack::new(slice);
         let mut str_ptr: [*const u8; SIZE] = [core::ptr::null(); SIZE];
-        let mut str_len: usize = 0;
+        let mut str_len: usize = 1;
         let mut rand_ptr: Option<*const u8> = None;
         let mut platform_ptr: Option<*const u8> = None;
         let mut base_platform_ptr: Option<*const u8> = None;
         let mut exec_fn_ptr: Option<*const u8> = None;
 
-        str_ptr[str_len] = core::ptr::null();
-        str_len += 1;
-
         for s in self.envp.iter() {
             if let Some(s) = s {
                 sp.push_l(0u8);
```

Reply:
yes, but needs a comment